gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/visitor/ExecutionVisitor.java

```diff
@@ -46,16 +44,17 @@
 
 /** Visitor that executes each QueryNode in the execution tree. */
 public class ExecutionVisitor implements Visitor<EntityResponse> {
-
-  private static final int THREAD_COUNT = 20;
-  private static final ExecutorService executorService = Executors.newFixedThreadPool(THREAD_COUNT);
+  private static final Logger LOG = LoggerFactory.getLogger(ExecutionVisitor.class);
 
   private final EntityQueryHandlerRegistry queryHandlerRegistry;
+  private final EntitiesRequest entitiesRequest;
   private final ExecutionContext executionContext;
-  private static final Logger LOG = LoggerFactory.getLogger(ExecutionVisitor.class);
 
   public ExecutionVisitor(
-      ExecutionContext executionContext, EntityQueryHandlerRegistry queryHandlerRegistry) {
+      EntitiesRequest entitiesRequest,
```

Comment:
Didnt understand the motivation for all these changes. Why did you have to pass the EntitiesRequest explicitly here where it is there in ExecutionContext?